conans/util/files.py

```diff
@@ -337,3 +338,33 @@ def exception_message_safe(exc):
         return str(exc)
     except Exception:
         return decode_text(repr(exc))
+
+
+def copy_symlink(pointer_src, src, dst, output=None):
+    """
+    Copy the symlink 'pointer_src' from the 'src' directory to the 'dst' one. It
+    will warn if the linked file is outside the 'src' directory
+
+    :param pointer_src: symlink inside 'src' directory
+    :param src: source directory
+    :param dst: destination directory
+    :param output:
+    :return:
+    """
+
+    linkto = os.readlink(pointer_src)
+    if not os.path.isabs(linkto):
+        linkto = os.path.join(os.path.dirname(pointer_src), linkto)
+
+    # Check if it is outside the sources
+    out_of_source = os.path.relpath(linkto, os.path.realpath(src)).startswith("../")
```

Comment:
⚠️ [Breaking?] It has changed from `.` to `../` so now we are also copying files/folders that starts with just a `.`.